julia/deps/build.jl

```diff
@@ -155,131 +156,66 @@ if !libmxnet_detected
   blas_path = Libdl.dlpath(Libdl.dlopen(Base.libblas_name))
   blas_vendor = LinearAlgebra.BLAS.vendor()
 
-  ilp64 = ""
-  if blas_vendor == :openblas64
-    ilp64 = "-DINTERFACE64"
-  end
+  USE_JULIA_BLAS = (blas_vendor in (:openblas, :openblas64))
+  @info "USE_JULIA_BLAS -> $USE_JULIA_BLAS"
 
-  FORCE_LAPACK = false
-  if blas_vendor == :unknown
-    @info("Julia is built with an unkown blas library ($blas_path).")
-    @info("Attempting build without reusing the blas library")
-    USE_JULIA_BLAS = false
-  elseif !(blas_vendor in (:openblas, :openblas64))
-    @info("Unsure if we can build against $blas_vendor.")
-    @info("Attempting build anyway.")
-    USE_JULIA_BLAS = true
-  else
-    USE_JULIA_BLAS = true
-    FORCE_LAPACK = true
-  end
-  @info("USE_JULIA_BLAS -> $USE_JULIA_BLAS")
-
-  blas_name = blas_vendor == :openblas64 ? "openblas" : string(blas_vendor)
-  MSHADOW_LDFLAGS = "MSHADOW_LDFLAGS=-lm $blas_path"
+  blas_name = occursin("openblas", string(blas_vendor)) ?  "open" : string(blas_vendor)
 
   #--------------------------------------------------------------------------------
   # Build libmxnet
   mxnet = library_dependency("mxnet", aliases=["mxnet", "libmxnet", "libmxnet.so"])
 
-  _prefix = joinpath(BinDeps.depsdir(mxnet), "usr")
+  _blddir = joinpath(BinDeps.depsdir(mxnet), "build")
   _srcdir = joinpath(BinDeps.depsdir(mxnet), "src")
   _mxdir  = joinpath(_srcdir, "mxnet")
-  _libdir = joinpath(_prefix, "lib")
-  # We have do eagerly delete the installed libmxnet.so
+
+  # We have do eagerly delete the build stuffs.
   # Otherwise we won't rebuild on an update.
-  run(`rm -f $_libdir/libmxnet.$(Libdl.dlext)`)
+  rm(_blddir, recursive=true, force=true)
```

Comment:
Previously it deleted the file libmxnet.* in the libdir 
Now we are deleting the entire build directory. What changed with that?

Reply:
well, the final result is the same -- making a fresh build, not increamental build.

This build script is simliar to Python's setup.py.
It's for creating fresh build for user.